### ISSUE TITLE
Add a method for retrieving the type information, including any restrictions of Capability

### ISSUE THREAD

**Issue by LanfordCai** [onflow/cadence#2324]
### Issue to be solved

We can use the iteration API to obtain the complete type information of a capability now:

```cadence
pub fun main(address: Address): Type? {
    let account = getAuthAccount(address)
    var t: Type? = nil
    account.forEachPublic(fun (path: PublicPath, type: Type): Bool {
        if path == /public/BasicBeastsCollection {
            t = type
            return false
        }
        return true
    }) 

    return t
}
```

command:

```
flow scripts execute ./scripts/get_type.cdc 0x5a16175a09403578  --network=mainnet
```

result:

```
Type<Capability<&A.de7a5daf9df48c65.BasicBeasts.Collection{A.1d7e57aa55817448.NonFungibleToken.Receiver,A.1d7e57aa55817448.NonFungibleToken.CollectionPublic,A.de7a5daf9df48c65.BasicBeasts.BeastCollectionPublic,A.1d7e57aa55817448.MetadataViews.ResolverCollection}>>()
```

**The result including all the restrictions of a Capability**

However, due to #2279 , this method may not be available at times. For example, use this command to execute the script above:

```
flow scripts execute ./scripts/get_type.cdc 0x6b79fb9b4a394a17  --network=mainnet
```

We get:

```
[Error Code: 1106] max interaction with storage has exceeded the limit (used: 20298577 bytes, limit 20000000 bytes)
goroutine 10915810169 [running]
```

The iteration API returns a `Type` that contains all restrictions of a `Capability`. I'm not sure how to obtain this info using other APIs. Could you add an API similar to `fun type(at path: StoragePath): Type?` for `CapabilityPath`?

### Suggested Solution

add an API similar to `fun type(at path: StoragePath): Type?` for `CapabilityPath`

**Comment by turbolent** [comment]
Makes sense, we might want to generalize this to `Path` for `AuthAccount`, and add a `fun type(at path: PublicPath): Type?` function for `PublicAccount`.

I don't know if you seen it, but there is a plan to replace the current linking-based capability API with [Capability Controllers](https://github.com/onflow/flow/pull/798). It might make sense to just add it there instead of improving the API which is likely going to get deprecated.

**Comment by LanfordCai** [comment]
I know about it, but I haven't looked at it closely yet. It's would be great to have this function in the new APIs
